src/types/event-types.ts

```diff
@@ -1,8 +1,17 @@
-import type { Event } from './event-log.js';
 import type { GenericMessageReply } from '../core/message-reply.js';
+import type { RecordsFilter } from './records-types.js';
 import type { AuthorizationModel, GenericMessage } from './message-types.js';
 import type { DwnInterfaceName, DwnMethodName } from '../enums/dwn-interface-method.js';
 
+export type EventsFilter = RecordsFilter & {
+  /** optional array of methods to filter */
+  method?: string[];
+  /** optional array of interfaces to filter */
+  interface?: string[];
```

Comment:
It can be used as a OneOf filter. We can discuss limiting the types of filters allowed by this interface (and it would be the same for the Subscription interface).

Original thought is someone could want to selectively filter for `Records` and `Permissions` interfaces by a particular `protocol` as an example.